wremnants/theory_corrections.py

```diff
@@ -690,67 +692,97 @@ def get_names(ihel):
+        logger.debug(
+            f"Making PDF uncertainty helper by helicity for PDF set {pdf_name}"
         )
-        logger.info(
-            f"Making PDF uncertainty helper for PDF set {pdf} ({pdf_name}) with renorm {pdf_renorm}"
+        pdf_renorm_name = (
+            pdf_name if theory_tools.pdfMap[pdf].get("renorm", False) else "pdf_uncorr"
         )
-        pdf_helper = make_pdf_uncertainty_helper_by_helicity(
+        pdf_helper = make_uncertainty_helper_by_helicity(
             proc=proc,
-            pdf=pdf_name,
-            pdf_renorm=pdf_renorm_name,
-            filename=common.data_dir
-            + f"/PDFs/w_z_gen_dists_maxFiles_m1_{pdf}_pdfByHelicity_skimmed.hdf5",
-            filename_renorm=common.data_dir
-            + f"/PDFs/w_z_gen_dists_maxFiles_m1_{pdf_renorm}_pdfByHelicity_skimmed.hdf5",
+            nom=pdf_name,
+            den=pdf_renorm_name,
+            filename=pdf_file_template.format(pdf=pdf),
+            var_ax_name="pdfVar",
             return_tensor=return_tensor,
         )
         if pdf_helper is not None:
             pdf_helpers[pdf_name] = pdf_helper
     return pdf_helpers
 
 
-def make_pdf_uncertainty_helper_by_helicity(
+def make_alphaS_uncertainties_helper_by_helicity(
     proc,
-    pdf,
-    pdf_renorm=None,
-    central_weights=False,
+    as_vars,
+    return_tensor=True,
+):
+    alphas_file_template = (
+        common.data_dir
+        + "/AlphaSByHelicity/w_z_gen_dists_{as_var}_maxFiles_m1_skimmed.hdf5"
+    )
+    as_helpers = {}
+    for as_var in as_vars:
+        logger.debug(
+            f"Making alphaS uncertainty helper by helicity for theory corr {as_var}"
+        )
+        as_helper = make_uncertainty_helper_by_helicity(
+            proc=proc,
+            nom=as_var,
+            den="theory_uncorr",
+            filename=alphas_file_template.format(as_var=as_var),
+            var_ax_name="vars",
+            return_tensor=return_tensor,
+        )
+        if as_helper is not None:
+            as_helpers[as_var] = as_helper
+    return as_helpers
+
+
+def make_uncertainty_helper_by_helicity(
+    proc,
+    nom,
+    den,
     filename=f"{common.data_dir}/angularCoefficients/w_z_gen_dists_maxFiles_m1_alphaSunfoldingBinning_helicity.hdf5",
-    filename_renorm=None,
+    filename_den=None,
+    central_weights=False,
     var_ax_name="pdfVar",
     return_tensor=True,
 ):
-    if filename_renorm is None:
-        filename_renorm = filename
+    """
+    Construct a CentralCorrByHelicityHelper from helicity cross sections stored in an hdf5 file.
+    """
+
+    if filename_den is None:
+        filename_den = filename
 
     # load helicity cross sections from file
     proc_map = {
         "Z": ("ZmumuPostVFP",),
         "W": ("WplusmunuPostVFP", "WminusmunuPostVFP"),
     }
 
-    def _collect_pdf_hist(pdf_name, filename):
-        hist_key = f"nominal_gen_{pdf_name}"
+    def _collect_hist(h, filename):
```

Comment:
Maybe use a more expressive variable instead of "h" "histogram_name" or so

Reply:
fixed